### PR TITLE
feat: Support Bazel platform mappings

### LINKED ISSUE #9360: Support bazel platforms
Skaffold uses `github.com/opencontainers/image-spec/specs-go/v1` and containerd strings like "linux/amd64" or "linux/arm64/v7" to represent platforms.

Bazel's platform system uses bazel targets to define platforms (https://bazel.build/extending/platforms). While there are common cpu and os definitions available, each Bazel project defines its own target platforms (!)

Currently, when skaffold invokes bazel, skaffold's platform information is not passed to Bazel. Bazel defaults to the host platform, unless the target skaffold is building explicitly specifies a platform. 

To allow a Bazel + Skaffold project to flexibly target multiple platforms, skaffold should provide a mechanism to invoke bazel with `--platforms` when cross-compiling, based on its own `--platform` setting.

### PR DESCRIPTION
**Pull request by aran**
This change introduces an artifact-level setting for configuring Bazel and
defines a platform mapping from Skaffold platforms to Bazel platforms.
The Bazel artifact builder then passes the --platforms flag to Bazel if applicable.

Fixes #9360.

### CODE REVIEW

**Comment by aran** [comment]
Looking for some feedback on whether I should drop this change, continue with some modifications to the approach, or continue as-is. Let me know! Thanks.

**Comment by aran** [comment]
@renzodavid9 This is the PR we discussed in the Skaffold office hours call at end of Feb. Would be great to get your feedback if you have a chance. Thanks!

File: pkg/skaffold/schema/latest/config.go [hunk 1]
```diff
@@ -240,6 +240,8 @@ type BuildConfig struct {
 	// Example: `["linux/amd64", "linux/arm64"]`.
 	Platforms []string `yaml:"platforms,omitempty"`
 
+	Bazel *BazelBuildConfig `yaml:"bazel,omitempty"`
```
Comment (renzodavid9) [inline pkg/skaffold/schema/latest/config.go:243]: I understand this is a proof of concept, but I'm wondering if this is something that can be move to the `BazelArtifact` struct instead? I think is better to have all the Bazel configs group there, and also I think we'll avoid to change some functions signatures. Thanks!

**Comment by renzodavid9** [comment]
Hey @aran,

I'll get familiar with the Bazel platforms to understand more about how they work. Also I'll check if we need to do changes in other parts of Skaffold in order to fully support this. Thanks!

File: pkg/skaffold/schema/latest/config.go (hunk 1 above)
Comment (aran) [reply pkg/skaffold/schema/latest/config.go:243]: I'll make this change. I think it might also help avoid edge cases with composability of skaffold configs down the line. Just to be explicit the downside of the tradeoff is this will mean a lot of duplicated config. A project might have many bazel artifacts each of which shares the same underlying platform definitions in a shared bazel workspace. Without a higher-level place to put workspace-global build settings, these will need to be copy-pasted and kept consistent in case of changes.

**Comment by aran** [comment]
If this looks ok from an API and rough approach standpoint, I can look into details, tests, compliance with Skaffold project standards, etc. 

One open edge case question would be how to handle if someone passes `--platforms` to `BazelArtifact` `BuildArgs` as well as `PlatformMappings`. 

Another one would be whether include any special handling for hierarchical skaffold platforms like `'linux/arm64/v8'`.

Let me know, thanks!

**Comment by aran** [comment]
@renzodavid9 I updated the PR title and top-level comment to meet project guidelines. This is now ready for review. 

 * Changes to the configuration are documented to spec.
 * Generated code is checked in.
 * Added unit test.
 * Updated `examples/bazel` and its duplicate in `integration/examples/bazel`. These I tested manually.
 * Filed an issue which this closes for project tracking.

File: pkg/skaffold/schema/latest/config.go [hunk 2]
```diff
@@ -1597,6 +1607,18 @@ type BazelArtifact struct {
...
+	// PlatformMappings configure the --platforms flag for `bazel build`
+	// based on the configured skaffold target platform.
+	// For example:
+	// ```yaml
+	// platforms:
+	//   - platform: linux/amd64
+	//     target: //platforms:linux-x86_64
+	//   - platform: linux/arm64
+	//     target: //platforms:linux-arm64
+	// ```
```
[the remark this answers is not in the document]
Comment (aran) [reply pkg/skaffold/schema/latest/config.go]: Done in #9363

File: pkg/skaffold/schema/latest/config.go [hunk 3]
```diff
@@ -331,6 +331,16 @@ type TaggerComponent struct {
...
 }
 
+// BazelPlatformMapping relates a skaffold platform (like 'linux/amd64')
+// to a workspace-specific bazel platform target (e.g. '//platforms:linux_amd64').
+type BazelPlatformMapping struct {
+	// Platform is the skaffold platform.
+	Platform string `yaml:"platform,omitempty"`
+
+	// BazelPlatformTarget is the bazel platform target to be passed to bazel's `--platforms` flag.
+	BazelPlatformTarget string `yaml:"target,omitempty"`
```
[the remark this answers is not in the document]
Comment (aran) [reply pkg/skaffold/schema/latest/config.go:341]: 👍

**Review by renzodavid9 — APPROVED** [review]
